[15.0][MIG] payroll

mtelahun opened the pull request:
Fixes: #19 
cc: #78

davejames committed:
payroll: rename module

this commit renames the module from hr_payroll to payroll. This is to avoid conflicting with the hr_payroll module included in Odoo Enterprise

davejames committed:
payroll: rename install setting for payroll_account module

Also removes links to localisations which have not yet been migrated. These can be re-added when the localisations are ported.

davejames committed:
payroll: add original pot file

davejames committed:
payroll: update new POT file

hilarak committed:
[IMP] payroll: black, isort, prettier

hilarak committed:
[MIG] payroll: Migration to 14.0

norlinhenrik committed:
[14.0][FIX] payroll: demo rule: python code with no indentation

norlinhenrik committed:
[14.0] payroll: new maintainer: appstogrow

nimarosa committed:
parent 830abfcae9440166206476c5b2c74932564483a1
author nicolasrsande <nicolasrsande@gmail.com> 1643135061 -0300
committer nicolasrsande <nicolasrsande@gmail.com> 1643668459 -0300

fix typo in leave calculation

default for worked days should not compute leaves, because leaves are calculated separately

we should not add data because it interfers with custom localization payroll modules

fix pre-commit

add demo data so the test can be executed

leaves should be computed in negative value to help creating salary rules

Add mantainer key

remove mantainer in payroll_account

14.0-minor-fixes

fix typo in leave calculation

we should not add data because it interfers with custom localization payroll modules

fix pre-commit

add demo data so the test can be executed

leaves should be computed in negative value to help creating salary rules

Add mantainer key

remove mantainer in payroll_account

14.0-payroll-minor-fixes

nimarosa committed:
add detail of the exeption in salary rule calculation

accept suggestion use repr() on the string

change indentation of error

nimarosa committed:
[14.0] [IMP] Code fields for python compute fields

Noel000 committed:
Translated using Weblate (Catalan)

Currently translated at 93.6% (266 of 284 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/ca/

Noel000 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (284 of 284 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/ca/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 86.9% (247 of 284 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (284 of 284 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (284 of 284 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (284 of 284 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

OCA-git-bot committed:
[IMP] update dotfiles [ci skip]

nimarosa committed:
[IMP] payroll: Add salary_rule_parameters functionality

fix build errors, the menu was loaded before the action

add menu to rule parameter

accept sugestions by appstogrow, will test in afternoon

fix typo

minor changes

[FIX] payroll: remove payslip view change. Will make it in another PR

nimarosa committed:
[IMP] payroll: take into account the contract start date in payslip calculation

nimarosa committed:
[IMP] payroll: recalculate worked_days and inputs if structure is changed

nimarosa committed:
[IMP] payroll: add amount_qty to paylip inputs

nimarosa committed:
[IMP] payroll: add missing states in status bar

nimarosa committed:
[IMP] payroll: ux changes in salary rules views

Commit:
[ADD] payroll: missing tag to compute method

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 78.1% (222 of 284 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/it/

nimarosa committed:
[IMP] payroll: add support for hr_public_holidays in work time calculation

nimarosa committed:
[IMP] payroll: remove empty lines

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (290 of 290 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

mtelahun committed:
[FIX] payroll unittest

The unittest for the payroll module is broken in several ways:
    1. It doesn't actually test anything in the payslip, just the states
       when various buttons are simulated
    2. The payslip is created programatically, so no payslip values are loaded
    3. The employee's contract is in "draft" state so even if it tried to
       load any values it wouldn't work because the contract isn't "open"
    4. It modifies a salary input line, but doesn't verify it worked. It
       doesn't work because of the above mentioned reasons.

To ensure a minimal amount of testing:
    o Make sure the employee's contract is in "open" state
    o Use the "Form" testing object when creating a payslip to ensure
      the onchange_employee() is run
    o Actually test the payroll calculations are done correctly

mtelahun committed:
[FIX] payroll - wrong quantity and rate during python computation of salary rule

A side effect of the way the payslip line quantity and rate are set when
amount_select is set to "code" causes them to never evaluate to zero. When they
evaluate to 0 the side effect causes them to be set to 1.0 and 100.0, respectively.

    "result_qty" in localdict and localdict["result_qty"] or 1.0
    "result_rate" in localdict and localdict["result_rate"] or 100.0

mtelahun committed:
[IMP] payroll: refactor of local dictionary handling in payslip for salary rule processing

This should make it easier for other modules to inherit and modify the
payroll module. This solution allows other modules to arbitrarily
inject their own values into salary rules. Furthermore, payroll managers
will be able to continue using '.' (dot) notation to refer to these new
values. Module implementers can inherit and overload the following methods:

get_payroll_dict()  - for providing values under the generic `payroll.` object
get_baselocaldict() - for providing 'first-level' objects that are
                      available directly to salary rules. For the sake
                      of clarity the object string should start with
                      the module name. Example: if the module is named
                      payroll_foo the values should be under `payroll_foo.`
get_contract_dict() - for values that depend on the current contract being evaluated
                      These values will be available under `this_contract.`

In the inherited methods the modules should return a python dictionary of
discreet values and/or nested objects using a BaseBrowsableObject.
Example:
If the module name is my_module:
    res = super().get_baselocaldict(contract, contracts)
    res.update({
        "my_module": BaseBrowsableObject(
            "foo": 10.0,
            "bar": 25.0,
            "baz": BaseBrowsableObject(
                {
                    value1: 100.0,
                    value2: 150.0,
                }
            )
        )
    })
    return res

The above values can be accessed in salary rules as:
my_module.foo
my_module.bar
my_module.baz.value1
my_module.baz.value2

mtelahun committed:
[IMP] payroll: remove dependency on demo data

mtelahun committed:
[IMP] payroll: add a test to show that if result_* are not set they get default values

nimarosa committed:
[IMP] payroll: Remove contract_advantage_template model and new module hr_contract_advantages

[ADD] Add new module hr_contract_advantages

[FIX] payroll_contract_advantages: Fix typo in view

[FIX] payroll_contract_advantages: Fix typo in import

[IMP] payroll: change security grups

[FIX] payroll: fix version

[FIX] payroll_contract_advantages: fix import typo

[FIX] payroll_contract_advantages: Fix security groups

[FIX] payroll_contract_advantages: Fix menu typo

[IMP] payroll_contract_advantages: Fix get_contract_dict function

nimarosa committed:
[IMP] payroll: payslip refactoring and functionality improvements

[IMP] payroll: refactor of the method get_worked_day_lines

[IMP] payroll: remove old implementation

[IMP] payroll: remove unused lines

[IMP] payroll: improve payslip views

[IMP] payroll: Add input line computation in onchange

[IMP] payroll: Add a button to refetch payslip data manually

[IMP] payroll: Add domain to contract in many2one tables

[IMP] payroll: Add more tracking in fields

[IMP] payroll: extracted the name computing method

[FIX] payroll: remove redundant method

[IMP] payroll: replace tracking=1 with tracking=True

[IMP] payroll: Add daterange widget for dates

[IMP] payroll: Refactor get_payslip_vals and add struct_id selection on runs

[IMP] payroll: In runs if struct_id is not selected we fallback to contract default

nimarosa committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (290 of 290 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

nimarosa committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (290 of 290 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es/

nimarosa committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (301 of 301 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

nimarosa committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (301 of 301 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es/

nimarosa committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (301 of 301 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

mtelahun committed:
[IMP] payroll: allow the use of positive values in leave days/hours calculation

The system parameter "payroll.leaves_positive" is used to control
whether the value for number_of_days/number_of_hours is positive or
negative. When False (the default) it wil be negative.

nimarosa committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (307 of 307 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

nimarosa committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (307 of 307 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es/

norlinhenrik committed:
[IMP] payroll: send email with payslip, fix payslip form view, better payslip line views

nimarosa committed:
[FIX] payroll: Fix Refactored _compute_name to contemplate more than one record

[FIX] payroll: fix _compute_name singleton error

mtelahun committed:
[FIX] payroll: use a custom working calendar so tests don't fail on weekends

nimarosa committed:
[IMP] payroll: Improve views, add child functionalities, allow computing name

[IMP] payroll: Improve views, add child functionalities, allow compute of name

[FIX] payroll: fix test expected values

[IMP] payroll: [IMP] payroll: Improve views, add child functionalities, allow compute of name

[IMP] payroll: Improve views, add child functionalities, allow compute of name

[IMP] payroll: Improve views, add child functionalities, allow compute of name

[IMP] payroll: Improve views, add child functionalities, allow compute of name

[IMP] payroll: Add test coverage for new child view and flag

[IMP] payroll: Improve views, add child functionalities, allow compute of name

[IMP] payroll: change tab name

[IMP] payroll: Add info alert in salary_rule form explaining functionality

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (304 of 304 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

nimarosa committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (304 of 304 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es/

nimarosa committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (313 of 313 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

nimarosa committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (313 of 313 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es/

nimarosa committed:
[IMP] payroll: hr_payslip refactoring and misc improvements

[IMP] payroll: Move and refactor category sum function (now can be inherited)

[IMP] payroll: _get_payslip_lines refactoring

[IMP] payroll: move basebrowsableobject to own file

[IMP] payroll: adapt compute method to new returned values

[IMP] payroll: Add compute_date and use "verify" unused state - improve views and filters

[IMP] payroll: Add compute_date and use "verify" unused state - improve views and filters

[IMP] payroll: misc changes

[FIX] payslip_change_state: Fix tests and exeptions acording new implementation

[ADD] hr_payroll_cancel: Add mantainer keys

[IMP] payroll: Change more detailes name for lines

[IMP] all: Change repo description and contributors file.

[IMP] payroll: change order

[IMP] payroll: minor changes

[IMP] payroll: minor changes

norlinhenrik committed:
[IMP] payroll: hr.payslip.line with date_from

norlinhenrik committed:
[IMP] payroll: hr.payslip & hr.payslip.run: order by id desc (most recently created on top)

mtelahun committed:
[IMP] payroll: default value for python code fields had lines in the comments that were not commented

mtelahun committed:
[IMP] payroll: make hr.salary.rule (form) condition and computation views more sane in Firefox

nimarosa committed:
[IMP] payroll: Add hook method for accesing localdict

[IMP] payroll: Add hook method for accesing localdict

nimarosa committed:
[IMP] payroll: extract and move _sum_salary_rule_category to payslip

nimarosa committed:
[IMP] payroll: add new objects documentation in help text

nimarosa committed:
[IMP] payroll: remove ** from baselocadict

nimarosa committed:
[IMP] payroll: remove sum from worked days table

nimarosa committed:
[IMP] payroll: Add object for accessing with dot notation to the dict

nimarosa committed:
[IMP] payroll: Merge payslip_cancel in base module and remove module

nimarosa committed:
[IMP] payroll: Merge payslip_change_state and remove module

nimarosa committed:
[IMP] payroll: add tests for new merged functionalities

[FIX] payroll: fix tests

[FIX] payroll: fix tests

[IMP] payroll: Add conditional button and computed method for setting

[IMP] payroll: Add conditional button and computed method for setting

[IMP] payroll: Move cancel test to payroll account

[IMP] payroll: Add hr_payslio_cancel refund sheet code

[FIX] payroll: tests change import

[FIX] payroll: Remove account in tests, will add another test in account module

[IMP] payroll_account: Add cancel tests for accounting module

[FIX] payroll_account: Test remove

[FIX] account_payroll: Fix tests

nimarosa committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (315 of 315 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

nimarosa committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (315 of 315 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es/

nimarosa committed:
[IMP] payroll: change button attrs

mtelahun committed:
payroll: silence pylint W7950(odoo-addons-relative-import)

mtelahun committed:
[FIX] payroll: regression introduced in #62

When calling _compute_payslip_line() call it on the payslip object and
not recordset 'self'.

mtelahun committed:
[IMP] payroll: refactor tests.common to prepare for more tests

mtelahun committed:
[IMP] payroll: re-factor _get_contracts() to work on recordset self

mtelahun committed:
[IMP] payroll: re-factor _get_salary_rules() to work on recordset self

mtelahun committed:
[IMP] payroll: rename and re-factor _get_payslip_lines() to work on recordset self

To preserve compatibility with third-party modules the re-factored
functionality is in a new method: get_lines_dict() and the
_get_payslip_lines() just calls this method. I expect that the new
multi-recordset function will be useful in reports: no more manually
adding up payslip lines from individual payslips!

mtelahun committed:
[IMP] payroll: re-factor onchange_struct_id() to work on recordset self

mtelahun committed:
[IMP] payroll: re-factor onchange_dates() and onchange_employee() to work on recordset self

mtelahun committed:
payroll: fix a couple of places where functional changes were inadvertently introduced during the refactor

norlinhenrik committed:
[IMP] payroll: rule _satisfy_condition() & _compute_rule() -> 2*4 methods

norlinhenrik committed:
[IMP] payroll: rule._compute_rule() returns dict of values

norlinhenrik committed:
[IMP] payroll: payslip.get_lines_dict() calls rule._reset_localdict_values()

norlinhenrik committed:
[IMP] payroll: payslip._compute_payslip_line() -> _compute_rule() & _get_line_dict()

norlinhenrik committed:
[IMP] payroll: rule._compute_rule_code() calls _get_rule_dict()

nimarosa committed:
[FIX] payroll_contract_advantages: Function name changed in recent commits

norlinhenrik committed:
[IMP] payroll: payslip._compute_payslip_line() & _get_lines_dict()

nimarosa committed:
[14.0] [FIX] payroll: Fix get_lines_dict() - skip assigning dict when the rule is blacklisted

nimarosa committed:
[14.0] [IMP] payroll: support adding python libraries or objects to localdict

nimarosa committed:
[14.0] [IMP] payroll: improve lines table view

nimarosa committed:
[14.0] [IMP] payroll: improve salary_rule and categories views and documentation

[14.0] [IMP] payroll: improve salary rules views and documentation

[14.0] [IMP] payroll: improve salary rules  and categories views

[14.0] [IMP] payroll: improve salary rules views and documentation

[14.0] [IMP] payroll: improve salary rules views and documentation

[IMP] payroll: fix repeated words

[14.0] [IMP] payroll: fix typos

nimarosa committed:
[14.0] [IMP] payroll: settings for payslip_lines edition and auto-compute on confirm

[14.0] [IMP] payroll: Add no compute on confirm setting

[14.0] [IMP] payroll: Add no compute on confirm setting

[14.0] [IMP] payroll: Add no compute on confirm setting

[14.0] [IMP] payroll: Add no compute on confirm setting

[14.0] [IMP] payroll: Add no compute on confirm setting

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 99.7% (363 of 364 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (364 of 364 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

mtelahun commented:
It looks like there's a build problem. Probably a stale config somewhere. Can someone take a look please?
@nimarosa @pedrobaeza

mtelahun commented:
I mean the build failures. I'll take care of the pre-commit failures.

mtelahun commented:
@nimarosa @pedrobaeza Sorry. It was my own fault for not checking the migration went properly.

nimarosa commented:
@mtelahun i think contract views changed a bit in v15.0 that's why build error is about a xpath not found.

nimarosa commented:
/ocabot migration payroll

nimarosa commented:
@mtelahun Please before finishing this do a rebase with 15.0 version since i pushed updated dotfiles to have v15.0 up to date.

norlinhenrik commented:
Can we replace `hr.rule.parameter` & `hr.rule.parameter.value` with a dependency on [base_time_parameter](https://github.com/OCA/server-tools/pull/2426)?

nimarosa committed:
[14.0] [FIX] payroll: Fix parent rules dependecy

nimarosa commented:
@norlinhenrik Hello Henrik, I think adding a dependency to a server-tools module will be bad for payroll. 

Most people don't download all oca repos in his instances so it will end up with an dependency error when trying to install payroll and not having the other module. I think it's good to maintain the less dependencies required here. 

Also, and this one is more personal, the way I see it the user who uses rule parameters expect to have the view inside payroll module. Doing this with an inherited module will be a little tricky because we need to filter all views by domain and add again the module field in base_time_parameter and think of a way to integrate it in payroll the way it's done now. 

In my instances I use time parameter to store specific payroll data that has to be easily changed by the user and also have some custom implementations around this model and no time to migrate them now to a new behavior or module. 

If you have a good way of doing this by leaving the current functionality intact you can propose a PR on the 14.0 branch and it will be migrated to 15. 

I'm not a fan of introducing big changes in migrations because it causes confusion and errors and will requiere migration scripts, that's why I wanted to do the big changes in 14.0 so we make modules maintenance more easy and straightforward. 

Hope you understand and feel free to suggest a PR if you can do it without affecting current functionality.

mtelahun commented:
@nimarosa All is green now. Sorry for the noise. This was the first migration I did in a *long* time.

During my migration I noticed that Odoo has removed the `code` field from hr.leave.type. So, I have reverted the payroll module to using the `name` instead.
Maybe, starting from 15.0 onward we need a new model to map hr.leave.type to a code in worked days. What do you think?

mtelahun commented:
> Can we replace `hr.rule.parameter` & `hr.rule.parameter.value` with a dependency on [base_time_parameter](https://github.com/OCA/server-tools/pull/2426)?

I think this is a separate issue and belongs in a different PR

nimarosa commented:
> During my migration I noticed that Odoo has removed the `code` field from hr.leave.type. So, I have reverted the payroll module to using the `name` instead.
> 
> Maybe, starting from 15.0 onward we need a new model to map hr.leave.type to a code in worked days. What do you think?

Hello @mtelahun. I hate when odoo changes things like this, but I think is has to be because enterprise payroll module didn't use that code because enterprise module is different (our module is better un my opinion). 

What do you think about inheriting hr.leave.type here in the payroll module and adding this field again? That way we can maintain current functionality that is key for the module, at least for me. I think using name in salary rules will generate problems. 

If you think that adding this field to leave type will fix it we can include it in this PR.

nimarosa commented:
@mtelahun also noticed you removed all string="" of a lot of fields. Why is that?

mtelahun commented:
Yes, that's a better solution. I'll do that.

On 14/10/2022 12:52, Nicolas Rodriguez Sande wrote:
>
>     During my migration I noticed that Odoo has removed the |code|
>     field from hr.leave.type. So, I have reverted the payroll module
>     to using the |name| instead.
>
>     Maybe, starting from 15.0 onward we need a new model to map
>     hr.leave.type to a code in worked days. What do you think?
>
> Hello @mtelahun <https://github.com/mtelahun>. I hate when odoo 
> changes things like this, but I think is has to be because enterprise 
> payroll module didn't use that code because enterprise module is 
> different (our module is better un my opinion).
>
> What do you think about inheriting hr.leave.type here in the payroll 
> module and adding this field again? That way we can maintain current 
> functionality that is key for the module, at least for me. I think 
> using name in salary rules will generate problems.
>
> If you think that adding this field to leave type will fix it we can 
> include it in this PR.
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/OCA/payroll/pull/81#issuecomment-1278773047>, or 
> unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AB4YOHGXVHIW4Y7XH3RXG4TWDEUPTANCNFSM6AAAAAAREGB4YU>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

mtelahun commented:
pre-commit errors:

- string attribute on `<tree>` view is no longer allowed

- in model creation string attribute on a field that is the same as the 
default string provided by the ORM are no longer allowed

On 14/10/2022 12:59, Nicolas Rodriguez Sande wrote:
>
> @mtelahun <https://github.com/mtelahun> also noticed you removed all 
> string="" of a lot of fields. Why is that?
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/OCA/payroll/pull/81#issuecomment-1278780572>, or 
> unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AB4YOHBLGVQPY4GRGJ4X37TWDEVJFANCNFSM6AAAAAAREGB4YU>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

nimarosa commented:
@mtelahun yes but you are removing string in model creation not on tree view. Correct me maybe I didn't understand

mtelahun commented:
Please re-read my reply. There are two places where the `string=` 
attribute raises errors.

1. in `<tree>` view

2. in model creation when the `string=` attribute is the same as the 
default one provided by the ORM

On 14/10/2022 15:28, Nicolas Rodriguez Sande wrote:
>
> @mtelahun <https://github.com/mtelahun> yes but you are removing 
> string in model creation not on tree view. Correct me maybe I didn't 
> understand
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/OCA/payroll/pull/81#issuecomment-1278940872>, or 
> unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AB4YOHHRVG6D64O6WXBI7E3WDFGXHANCNFSM6AAAAAAREGB4YU>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

nimarosa commented:
@mtelahun great I understand now. 
Please tell me when it's ready so I review and merge.

Also if you can review the fix I made in 14.0 we can merge that first and include this change in following versions too.

mtelahun commented:
@nimarosa it's ready for review.
No problem, it can wait for #83

nimarosa commented on payroll/models/hr_payslip.py:
```diff
@@ -0,0 +1,803 @@
+                {
+                    "name": holiday.holiday_status_id.name or _("Global Leaves"),
+                    "sequence": 5,
+                    "code": holiday.holiday_status_id.name or "GLOBAL",
```
@mtelahun Okay so here you continue using name. Also i think you have to add the new file adding the field to hr.leave.type.

I would prefer we include this change in this PR because its a crucial functionality of the module.

mtelahun replied:
Done

nimarosa committed:
[14.0] [FIX] payroll: Add tests about parent/child rules

[14.0] [FIX] payroll: Add tests about parent/child rules

[14.0] [FIX] payroll: Add tests about parent/child rules

[14.0] [FIX] payroll: Add tests about parent/child rules

[14.0] [FIX] payroll: Add tests about parent/child rules

[14.0] [FIX] payroll: Add tests about parent/child rules

[14.0] [IMP] payroll: Add rule identification on error

[14.0] [FIX] payroll: Add tests about parent/child rules

[14.0] [IMP] payroll: Rollback rule identification on error

[14.0] [FIX] payroll: Add tests about parent/child rules

nimarosa committed:
[14.0] [ADD] hr_payroll_period

[14.0] [ADD] hr_payroll_period: Add mantainers key

[14.0] [ADD] hr_payroll_period: Change manifest category

[14.0] [ADD] hr_payroll_period

[14.0] [ADD] hr_payroll_period

[14.0] [IMP]  hr_payroll_period: Improve views

[14.0] [IMP] hr_payroll_period: Improve views and misc

[14.0] [IMP] hr_payroll_period: Improve views and misc

[14.0] [IMP] hr_payroll_period: Improve views and misc

nimarosa committed:
[14.0] [IMP] payroll: Add employee in rule exceptions

[14.0] [IMP] payroll: Add employee in rule exceptions

norlinhenrik committed:
[IMP] payroll: hr.salary.rule is readonly for payroll user

norlinhenrik committed:
[IMP] payroll: depends on base_time_parameter

[FIX] payroll: rule_parameter

[MIG] payroll: depends on base_time_parameter

[FIX] payroll migration

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (364 of 364 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

mtelahun commented:
@nimarosa @apps2grow Please review. It's ready now and includes recent changes in 14.0

nimarosa commented:
@mtelahun Hello, we will wait to close/merge #87 that has to be changed to create a new module and remove the parameters functionality from payroll. Then we can rebase and we start merging the migrations.

norlinhenrik committed:
[ADD] payroll_rule_time_parameter (instead of payroll) depends on base_time_parameter

norlinhenrik committed:
[IMP] payroll: history

norlinhenrik committed:
[IMP] payroll: hr.payslip.run with company_id

[FIX] payroll: fix views

norlinhenrik committed:
[IMP] payroll: code & category are not required

norlinhenrik committed:
[IMP] payroll: settings descriptions, show_details_by_salary_rule_category

norlinhenrik committed:
[IMP] payroll: required fields, removed details_by_salary_rule_category

norlinhenrik committed:
[IMP] payroll: hide_invisible_lines (who don't appear on payslip)

norlinhenrik committed:
[IMP] payroll: New setting: Require code and category

norlinhenrik committed:
[IMP] payroll: test rule code in payslip line

nimarosa committed:
[IMP] payroll: remove parameters code

nimarosa committed:
[14.0] [IMP] payroll: add localdict pure payslip object and rename payslips browsable object

nimarosa committed:
[14.0] [IMP] payroll: remove old tests from payroll module

nimarosa committed:
[14.0] [IMP] payroll: improve comments and documentation of available objects

nimarosa committed:
[14.0] [IMP] payroll: change manifest category

[14.0] [IMP] payroll: change manifest category

[14.0] [IMP] payroll: add migration for new payslip and payslips objects

[14.0] [IMP] payroll: add migration for new payslip and payslips objects

[14.0] [IMP] payroll: add migration for new payslip and payslips objects

norlinhenrik committed:
[IMP] payroll: removed required=False, improved text descriptions

norlinhenrik committed:
[IMP] payroll: test rule and category with and without code

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (359 of 359 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

mtelahun committed:
[IMP] payroll: black, isort, prettier

mtelahun committed:
[MIG] payroll: Migration to 15.0

mtelahun commented:
@nimarosa @apps2grow Please review and let's merge. This incorporates the latest changes.

nimarosa commented:
@mtelahun for me it's okay. Just done a code review and commit history looks okay. 

Did you tested it yet? Like functional.

norlinhenrik reviewed (approved):
I successfully computed a payslip for Marc Demo.

nimarosa commented:
Okay let's go. 

I have a few PRs that I might add to 14.0 but we can migrate them later.

/ocabot merge nobump

mtelahun commented:
@nimarosa Yes, I tested it. Thanks for merging.


Mike.

On 11/11/2022 01:31, Nicolas Rodriguez Sande wrote:
>
> Okay let's go.
>
> I have a few PRs that I might add to 14.0 but we can migrate them later.
>
> /ocabot merge nobump
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/OCA/payroll/pull/81#issuecomment-1310985900>, or 
> unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AB4YOHCMPD2LPCSOLNZUJR3WHVZV7ANCNFSM6AAAAAAREGB4YU>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>
